src/Entity/EntityCustomField.php

```diff
@@ -159,4 +164,20 @@ public static function getStorageFieldName()
     {
         return 'entityValue';
     }
+
+    public function __clone()
+    {
```

Comment:
at least until short ago, we were not allowed to overwrite clone like this. ([in doc for 2.7](https://github.com/doctrine/doctrine2/blob/2.7/docs/en/cookbook/implementing-wakeup-or-clone.rst))
But this seems to be valid no longer at least form doctrine **master** branch [doctrine/doctrine2@3411950](https://github.com/doctrine/doctrine2/commit/3411950c6cf5d8c63005ed60ba31b05ab4236ebd#diff-a806de2d9877d245fd96d645122a35df)
But since I do not know if it is valid for 2.6, I suggest to do it the old way:
```php
public function __clone() {
    if ($entity->id) {
        ... // do special cloning
    }
}
```

Reply:
good point, will change this accordingly. note that $entity->id is not available, must bei $entity->getId()